pkg/apis/storage/types.go

```diff
@@ -310,6 +327,36 @@ type CSIDriverSpec struct {
 	VolumeLifecycleModes []VolumeLifecycleMode
 }
 
+// FSGroupPolicy specifies if a CSI Driver supports modifying
+// volume ownership and permissions of the volume to be mounted.
+// More modes may be added in the future.
+type FSGroupPolicy string
+
+const (
+	// Heuristic indicates that each volume should be examined
+	// to determine if the volume ownership and permissions
+	// should be modified.
+	// This mode should be defined if it's expected that the
+	// fsGroup may need to be modified depending on the
+	// pod's SecurityPolicy.
+	HeuristicFSGroupPolicy FSGroupPolicy = "Heuristic"
+
+	// Supported indicates that the volume ownership and permissions will
+	// be modified each time the volume is mounted. Volumes
+	// created by this CSI Driver will have their filesystem updated
+	// to match the defined fsGroup on each mount.
+	// This mode should be defined if the fsGroup is expected
+	// to always change on mount.
+	SupportedFSGroupPolicy FSGroupPolicy = "Supported"
```

Comment:
The documentation should say - 

"Supported indicates that CSI driver supports volume ownership and permission change via fsGroup and Kubernetes may  use fsGroup to change permissions and ownership of the volume to match user requested fsGroup in pod"

Reply:
I've taken this and also added that it's the fsGroup from the Pod's SecurityPolicy